src/Features/Core/Portable/GenerateConstructorFromMembers/AbstractGenerateConstructorFromMembersCodeRefactoringProvider.cs

```diff
@@ -202,6 +202,12 @@ public override Task ComputeRefactoringsAsync(CodeRefactoringContext context)
                 return null;
             }
 
+            // We shouldn't offer a refactoring if the compilation is invalid.
+            if (semanticModel.Compilation.GetTypeByMetadataName("System.ArgumentNullException") is null)
```

Comment:
Is this specific type important, or just being used as a sentinel for "do we have a framework of some kind"?